src/processor.cpp

```diff
@@ -0,0 +1,394 @@
+#include "game.h"
+#include "map.h"
+#include "mapdata.h"
+#include "generic_factory.h"
+#include "messages.h"
+#include "json.h"
+#include "vehicle.h"
+#include "processor.h"
+
+#include <string>
+
+//process namespace
+
+namespace
+{
+    struct process_data;
+    using process_id = string_id<process_data>;
+
+    struct process_data {
+
+        process_data() :
+            was_loaded(false) {};
+
+        process_id id;
+        std::string name;
+        std::vector<std::string> components;
+        std::string output;
```

Comment:
Try to use `itype_id` instead of raw string where it makes sense. It's not a strict requirement, but one day we may want to turn item type ids into proper ids and then it will be helpful not to have to redefine types. It also makes reading it easier - you instantly know that you're looking at an item type rather than something else.